src/System.Net.Security/src/System/Net/_SSPIWrapper.cs

```diff
@@ -0,0 +1,873 @@
+            var unmanagedBuffer = new Interop.Secur32.SecurityBufferStruct[input.Length];
+
+            fixed (Interop.Secur32.SecurityBufferStruct* unmanagedBufferPtr = unmanagedBuffer)
+            {
+                sdcInOut.UnmanagedPointer = unmanagedBufferPtr;
+                GCHandle[] pinnedBuffers = new GCHandle[input.Length];
+                byte[][] buffers = new byte[input.Length][];
+                try
+                {
+                    for (int i = 0; i < input.Length; i++)
+                    {
+                        SecurityBuffer iBuffer = input[i];
+                        unmanagedBuffer[i].count = iBuffer.size;
+                        unmanagedBuffer[i].type = iBuffer.type;
+                        if (iBuffer.token == null || iBuffer.token.Length == 0)
+                        {
+                            unmanagedBuffer[i].token = IntPtr.Zero;
+                        }
+                        else
+                        {
+                            pinnedBuffers[i] = GCHandle.Alloc(iBuffer.token, GCHandleType.Pinned);
+                            unmanagedBuffer[i].token = Marshal.UnsafeAddrOfPinnedArrayElement(iBuffer.token, iBuffer.offset);
+                            buffers[i] = iBuffer.token;
+                        }
+                    }
+
+                    // The result is written in the input Buffer passed as type=BufferType.Data.
+                    int errorCode;
+                    switch (op)
+                    {
+                        case OP.Encrypt:
+                            errorCode = SecModule.EncryptMessage(context, sdcInOut, sequenceNumber);
+                            break;
+
+                        case OP.Decrypt:
+                            errorCode = SecModule.DecryptMessage(context, sdcInOut, sequenceNumber);
+                            break;
+
+                        case OP.MakeSignature:
+                            errorCode = SecModule.MakeSignature(context, sdcInOut, sequenceNumber);
+                            break;
+
+                        case OP.VerifySignature:
+                            errorCode = SecModule.VerifySignature(context, sdcInOut, sequenceNumber);
+                            break;
+
+                        default: throw NotImplemented.ByDesignWithMessage(SR.net_MethodNotImplementedException);
+                    }
+
+                    // Marshalling back returned sizes / data.
+                    for (int i = 0; i < input.Length; i++)
+                    {
+                        SecurityBuffer iBuffer = input[i];
+                        iBuffer.size = unmanagedBuffer[i].count;
+                        iBuffer.type = unmanagedBuffer[i].type;
+
+                        if (iBuffer.size == 0)
+                        {
+                            iBuffer.offset = 0;
+                            iBuffer.token = null;
+                        }
+                        else
+                            checked
```

Comment:
Please add braces here, when the statements un-indent extra after the closing brace it's a bit confusing.